docs/develop-guideline.md

```diff
@@ -0,0 +1,120 @@
+# 1. Overview
+
+The `enclave-agent` runs on a SGX-based LibOS. In the runtime, the memory and the mounted filesystem used by the enclave agent is encrypted by the SGX. 
+
+The supporting LibOS includes
+- [occlum](https://github.com/occlum)
+    - the encrypted filesystem is called by `unionfs`
+- [gramine](https://github.com/gramineproject/gramine)
+
+The dependencies installation process can be found in the [develop-env-install.md](develop-env-install.md).
+
+
+## Image Management
+
+The enclave agent plays the role of the container image management. It is responsible for four basic missions, including
+- Pulling: it pulls the index, manifest, config and layers of a image, which is dependent on the crate [image-rs](https://github.com/confidential-containers/image-rs)
+- Unpacking: it decrypts and decompresses the layers of a image if necessary. The decryption is dependent on
+the crate [ocicrypt-rs](https://github.com/confidential-containers/ocicrypt-rs).
+- Storing: it saves the unpacking layers on an encrypted filesystem
+- Mounting: it merges layers to a rootfs for a application container and mounts the rootfs to a mounting point
+
+
+## Security Management
+
+The enclave agent need to satisfy security requirements, including
+- attestation
+    - remote attestation: it provides security information for the remove 
+    - local attestation
+    - the crate [attestation-agent](https://github.com/confidential-containers/attestation-agent) supports a native attesation client
+- signature verification: 
+    - some images are signed by the pulisher or the store or both, it need to verify these signatures
+    - the crate [attestation-agent](https://github.com/confidential-containers/attestation-agent) supports a native signature client 
+
+# 2. Development Guide
+
+## Occlum-based Enclave Agent
+
+### *Dependencies*
+
+#### **rats-tls**
```

Comment:
this integration work is still not complete so I think it's not yet the right time to add docs for it.

Reply:
The section is removed and will be added in the PR related to `eaa kbc`.